projects/components/src/select/select.component.ts

```diff
@@ -204,6 +210,27 @@ export class SelectComponent<V> implements AfterContentInit, OnChanges {
     }
   }
 
+  public getPrefixIcon(
+    selectedOption: SelectOptionComponent<V> | undefined,
+    triggerIcon: string | undefined
+  ): string | undefined {
+    if (selectedOption !== undefined && selectedOption?.icon !== undefined) {
+      return selectedOption.icon;
+    }
+
+    if (selectedOption !== undefined) {
```

Comment:
if selected option is defined then we are returning undefined for the icon. I didn't understand this.

did you mean `selectedOption=== undefined` ?

Reply:
it's saying hide the trigger icon if anything is selected which matches the test logic. honestly not sure how we expect it to be used, but the logic can convey the current intent better IMO by collapsing the whole function to:
```ts
  return selectedOption ? selectedOption.icon : triggerIcon;
```

Reply:
fixed this.